phat/contracts/lego_rs/lib.rs

```diff
@@ -0,0 +1,152 @@
+#![cfg_attr(not(feature = "std"), no_std, no_main)]
+
+#[macro_use]
+extern crate alloc;
+
+#[ink::contract]
+mod lego {
+    use crate::raw_bytes::RawBytes;
+    use alloc::string::String;
+    use alloc::vec::Vec;
+    use ink::env::Error;
+    use pink::{info, warn};
+    use scale::Encode;
+    use serde::Deserialize;
+    use this_crate::{version_tuple, VersionTuple};
+
+    type Result<T, E = Error> = core::result::Result<T, E>;
+
+    #[derive(Debug, Deserialize)]
+    #[serde(tag = "cmd", content = "config")]
+    #[serde(rename_all = "camelCase")]
+    enum Action {
+        Call {
+            #[serde(with = "deserialize_hex")]
+            callee: [u8; 32],
+            selector: u32,
+        },
+        Log,
+    }
+
+    mod deserialize_hex {
+        use super::*;
+
+        pub(super) fn deserialize<'de, D>(deserializer: D) -> Result<[u8; 32], D::Error>
+        where
+            D: serde::Deserializer<'de>,
+        {
+            let s = String::deserialize(deserializer)?;
+            let mut bytes = [0u8; 32];
+            hex::decode_to_slice(s.as_str().trim_start_matches("0x"), &mut bytes)
+                .map_err(serde::de::Error::custom)?;
+            Ok(bytes)
+        }
+    }
+
+    #[ink(storage)]
+    pub struct Lego {}
+
+    impl Lego {
+        #[ink(constructor)]
+        pub fn default() -> Self {
+            Self {}
+        }
+
+        #[ink(message)]
+        pub fn version(&self) -> VersionTuple {
+            version_tuple!()
+        }
+
+        #[ink(message)]
+        pub fn run(&self, actions: String) -> bool {
+            info!("lego_rs: actions={}", actions);
```

Comment:
We shall not log actions here. E.g. in the future we may transfer some sensitive values among actions.

Reply:
Aagin, this PR is for special usage. When we do the full implmementation, the log would be controled by the `debug` flag in the workfow manifest in the future. As it defined in #3.

Reply:
Created an issue #21 to record this.